### PR TITLE
Docs: Gitter -> Discord URL (refs #13039)

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [x] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Replaced all links to the Gitter chat with a link to Discord (https://eslint.org/chat, to make it easy to change later)

#### Is there anything you'd like reviewers to focus on?

Did I miss any references to Gitter?

### CODE REVIEW

**Comment by nzakas** [comment]
Depends on: https://github.com/eslint/website/pull/730

File: .github/ISSUE_TEMPLATE/NEW_RULE.md [hunk 1]
```diff
@@ -8,18 +8,20 @@ assignees: ''
 ---
 
 <!--
-    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).
+    ESLint adheres to the Open JS Foundation Code of Conduct:
+    https://eslint.org/conduct
 
-    This template is for new rule proposals. If you are proposing a new rule, please continue on. If you are here for another reason, please see below:
+    This template is for bug reports. If you are here for another reason, please see below:
```
Comment (kaicataldo) [inline .github/ISSUE_TEMPLATE/NEW_RULE.md:14]: Is this change accurate? It looks like all the templates now say they are intended for bug reports.

Comment (nzakas) [reply .github/ISSUE_TEMPLATE/NEW_RULE.md:14]: Oops, copy-paste error! Nice catch.

**Comment by nzakas** [comment]
Okay, copy-paste errors have been resolved. My IDE fu is weak. :(

**Pull request merged**